Petrov 2.0

CakeQ opened the pull request:
🆑 Cakey
maptweak: Changed the Petrov into a two-story shuttle capable of housing NT and their desire to chase Phoron. Moved the RD's office to the Petrov, along with adding an RnD lab, misc lab, locker rooms, equipment storage and general living areas.
maptweak: Re-mapped the ERT shuttle to fit into its' new dock.
tweak: The Research Director and NTL can now use the ID program to modify accesses within their own department. Petrov accesses are now only modifiable by the Research Director and NTL
tweak: Fixes the commissarry shutter access requirement.
/🆑

![image](https://user-images.githubusercontent.com/17550641/45156235-8ad27980-b1d5-11e8-882e-6fa4345c9915.png)

![image](https://user-images.githubusercontent.com/17550641/45156254-945be180-b1d5-11e8-966e-1136bfa685fc.png)

![image](https://user-images.githubusercontent.com/17550641/45156296-adfd2900-b1d5-11e8-9e61-c9075c0dea86.png)

![image](https://user-images.githubusercontent.com/17550641/45157202-57451e80-b1d8-11e8-828c-b2d413b58098.png)

![image](https://user-images.githubusercontent.com/17550641/45156544-72af2a00-b1d6-11e8-8fc1-2887fb0adcd0.png)

![image](https://user-images.githubusercontent.com/17550641/45156705-d5082a80-b1d6-11e8-96b8-1d7f615f67d3.png)



There's an empty room. Leaving it empty unless someone has any suggestions for it.

- [x] Add new access to NT boys.
- [x] Give RD access to NT ID changing.
- [x] Give security guards access to the NT network.
- [x] Test the Petrov.

[Thread](https://forums.baystation12.net/threads/petrov-2-0.6965)

CakeQ committed:
Petrov 2.0

Petrov 2.0 second pass

Petrov 2.0 third pass

Petrov 2.0 fourth pass

update unit test

Petrov 2.0 fifth pass

Petrov 2.0 second pass

ghost commented:
Only gripe is the loss of a NTSec checkpoint on D1. Had one on both the old Petrov and the old D1, so this is just going to leave NT with just the one on the Petrov.

Are D1 lockdown shutters still intact?

HeyBanditoz commented:
Why is the middle test chamber 3x3 instead of 4x3 like the others?

Faustico commented:
NT sec definitely needs a checkpoint on deck one.

With pretty much everything being in the Petrov now what's stopping NT from just piling in and undocking every time the Torch goes above code blue? Also this is gonna make anomaly research a LOT more disruptive.

ghost commented:
I'm OK with higher-risk anomaly research. EMPs and radiation may be a bit of a bother, but anom research should be one of those things where people actually take complete precautions versus "remove from crate smack with stun baton wait for result." I feel this would nudge it toward a direction where anom would actually have to take inadvertently harming others into account.

Sbotkin reviewed (commented):
1. Where are the clones?
2. Two offices for one RD is too much.
3. One console for the checkpoint is enough.
4. Where is it docked?
5. No lounge for NT staff.
6. Excavation equipment is located on the D4, no need to double.
7. Do multiple z-level shuttles even work?
8. Should be a checkpoint on D1.
9. Two fabrication labs is too much. We aren't going to mass produce weaponry, are we?
10. With **that** much removed from the Torch, at that point it would be better to move everything NT related to the Petrov.
11. We tend to use other shuttles from IC standpoint when we overhaul them (see Calypso -> Charon). Consider a rename. I'd stick to cyrillic name, sounds cool and sciency.

comma commented:
Not sure if Cakey is alive so going to drop my deuce while I'm in the area


>what's stopping NT from just piling in and undocking every time the Torch goes above code blue?

Nothing. Go ahead.

>anomaly research

Barely happens, and was even too safe with detached lab.

>Two offices for one RD is too much.

Where's second one?

>Where are the clones?

Across hallway from cryo

>Do multiple z-level shuttles even work?

Theoretically

>With that much removed from the Torch, at that point it would be better to move everything NT related to the Petrov.

That's the plan in longer run I think, shipside labs be EC-owned (with nt access), Petrov labs pure NT

>Should be a checkpoint on D1.

Not really NT labs anymore, cue no checkpoint. Maybe general sec one.

>Two fabrication labs is too much. We aren't going to mass produce weaponry, are we?

Going to be on separate nets, not sharing data. Sipside one is mostly for robotics

CakeQ commented:
Oh, yeah, the Bridge RD office is being removed.

Faustico commented:
I don't like the idea of moving NT offship, for the obvious reason that it will decrease interaction with the rest of the Torch and by extension it will decrease the likelyhood of it being affected (intentionally or not) by antag actions.

comma commented on code/modules/research/rdconsole.dm:
```diff
@@ -797,11 +797,14 @@ won't update every console in existence) but it's more of a hassle to do. Also,
-	name = "core fabricator console"
-	id = 1
+/obj/machinery/computer/rdconsole/petrov
+	name = "petrov fabricator console"
```
torch specific shit should be defined in maps/torch folder

CakeQ replied:
Resolved.

Nerev4r commented:
Is it capable of going places, and landing?

comma commented:
## UNIT_TEST ##: [UNIQUENESS: Access Datums Shall Be Unique]: Ids - 200 - The following entries have the same value: /datum/access/crate_cash and /datum/access/petrov
## UNIT_TEST ##: [UNIQUENESS: Access Datums Shall Be Unique]: Ids - 201 - The following entries have the same value: /datum/access/merchant and /datum/access/petrov_helm

200 and 201 are used already

## UNIT_TEST ##: !!! FAILURE !!! [POWER: Each area should have at most one APC.]: Duplicated APCs in area: Charon - Cargo Bay. #1: The Charon - Cargo Bay APC (/obj/machinery/power/apc) ([0x20015d6]) (the deck) (112,107,1) (/turf/simulated/floor/tiled/dark)  #2: The Charon - Cargo Bay APC (/obj/machinery/power/apc/high) ([0x2001958]) (the deck) (114,112,1) (/turf/simulated/floor/tiled/dark)

two APCs in charon???

https://travis-ci.org/Baystation12/Baystation12/jobs/423190053#L922-L925
disposal derps

https://travis-ci.org/Baystation12/Baystation12/jobs/423190053#L940
cable overlap

https://travis-ci.org/Baystation12/Baystation12/jobs/423190053#L942-L951
areas that either need missing crap or to be added to exemptions list

https://travis-ci.org/Baystation12/Baystation12/jobs/423190053#L1001-L1005
unused areas

https://travis-ci.org/Baystation12/Baystation12/jobs/423190053#L1007-L1011
something's fucked

afterthought2 commented:
Someone will need to confirm that the multi-z shuttle moves properly between landmarks. While in theory shuttles can handle being multi-z, this hasn't been used previously. Of particular concern: if the target z level (n) is not connected to z level (n+1), or (n+1) doesn't exist, what happens? What if (n+1) is an entirely unrelated z level, e.g. the admin area?

Textor44 commented:
Why is the xenobotany lab so much smaller after this? With all the extra lab space, it should be grown, not shrunk. Most of deck 2 is already wasted space, do we need more wasted space on deck 1 now?

sick-trigger commented:
pros: petrov moving means observation dome's view of space no longer obscured

cons: the lounge's view of space is now obscured by petrov!!

VNLoverMay commented:
Not a fan of the new xenobotany lab, you've halved the number of available trays for no good reason and made the seed storage more out of the way than it already is. Why the hell is it across the hallway? Come on, knock out some of that maintenance and give it some more space, don't make it so much smaller when there's no reason to.

Maveriknight commented:
thing to keep in mind, making a two story petrov able to go to away sites would likely be bad since most away sites are 1 Z above ground at best.

CakeQ commented:
@afterthought already confirmed by chinsky

BlueNexus commented:
I don't like leaving such a large abandoned area on the torch. Any idea for what could be put there instead?

I'm also not sure I like moving NT off the torch, since it'll reduce interaction.

comma commented:
![](https://i.imgur.com/7Cv4PMt.png)
Random strip of space area on lower deck

Old petrov areas are still floating in space

Shuttle datum is still trying to move old area, list of aras needs to be updated
Would be easier to do
```
/datum/shuttle/autodock/ferry/petrov/New(_name, var/obj/effect/shuttle_landmark/initial_location)
	shuttle_area = subtypesof(/area/shuttle/petrov)
	..()
```
in torch_shuttles.dm

CakeQ commented:
Made xenoflora bigger:
![image](https://user-images.githubusercontent.com/17550641/44945852-01900100-ade9-11e8-88fd-78ae9d66a716.png)

Neon1ight commented:
I agree that moving the most important NT assets off-ship will increase the insularity of NT, and it's already very much cut off from the rest of the ship at large oftentimes. At the same time, though...

>That's the plan in longer run I think, shipside labs be EC-owned (with nt access), Petrov labs pure NT

I would fucking love it if I got to play EC science with the shipboard lab. Xenobotany, I will return to you...

GracieGrace0 commented:
I definitely do not like the idea of moving NT completely off of Torch. SCG and NT are supposed to be working together and moving them off the ship will just further reduce that.

Nerev4r commented:
No matter what we do, NT players will always remain insular. This paves the way for EC science, something actually needed.

Textor44 commented:
I still feel like there's a lot of wasted space here. The unused room should be turned back into a break room/lounge area. If not for NT only,, then in general. Or, move the R&D lab over more and put in a checkpoint again.

babydoll commented:
Any further talk about 'NT players' as if they aren't part of this community will net a warning. I'm really sick of it. Develop with everyone in mind. This is your duty as a contributor. It's not you vs. them.

babydoll commented:
The map diff bot isn't working so you'll need to provide images of all your changes across all the maps

Dopupureganjiru commented:
I think this is one of the worst ideas i have ever seen proposed and does nothing to encourage RP, and only to isolate and promote bias against the people who actually use the space. I am tired of seeing PR's that treat science / R&D with contempt either subtly, or overtly.  This seems to do both. It's apalling, and I hope this PR gets halted or deeply reviewed. 

Most of the points have already been brought up:
Moving things off ship discourages interaction. 
NTsec needs a checkpoint.
Moving /removing the RD's office further isolates RP.
Anomalies arent really securable, they bypass their crates and have AOE effects often enough that putting them in any populated area could destroy a round for that entire section. Would need to see how frequently it actually occurs. 
if the images of the redesigned D1 area are accurate, the inflate areas that are rarely used, eliminate areas that promote rp, and shrinks the over-all department for no clear reason. 

Making the Petrov what amounts to a mini-torch is tantamount to making the  'Sovereign NT' meme a reality, and I see no positive consequences of this design. Negative IC impact from reduced interaction. VERY negative ooc impact from reinforcing the divisive idea, and no mechanical game benefit from any angle except being able to eject the entire science division if something goes wrong, which can largely already be done if the Petrov is used as intended, as it currently exists. 

I feel what Sabiram pointed out has been the norm for development ideas for months and this is just the latest attempt to further push NT out of the play area, or out of play entirely. If this nets me a warning I'll take it, but I'm sick of seeing stuff like this that amounts to 'remove NT' while claiming to alter or add things. The redesign reeks of contempt in my opinion, and is a lot of work put in to making life worse for a few people.

HeyBanditoz commented:
All the interaction nt really needs with the crew is rnd/ fabrication so engineering can get advanced capacitors and robutts can do shit

On September 1, 2018 8:18:39 PM MDT, Dopupureganjiru <notifications@github.com> wrote:
>I think this is one of the worst ideas i have ever seen proposed and
>does nothing to encourage RP, and only to isolate and promote bias
>against the people who actually use the space. I am tired of seeing
>PR's that treat science / R&D with contempt either subtly, or overtly. 
>This seems to do both. It's apalling, and I hope this PR gets halted or
>deeply reviewed. 
>
>
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/Baystation12/Baystation12/pull/22905#issuecomment-417900077

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

Dopupureganjiru commented:
This attitude is precisely what i'm talking about though. There's more to interaction than the mechanical benefit to your favorite departments. This entire redesign is built around the idea of downsizing the NT presence ON the torch itself. 

When R&D is working well, you can see improvements in robotics, medical machines / surgery tools, prospecting, most synths can benefit from improved chargers, experimental welders are sometimes pretty handy, and since the inclusion of inducers - security, engineering, and exploration can get portable recharging tools. Even the kitchen can benefit from an improved biogenerator for meat and milk. There's a multitude of ways a creative scientist can help out around the ship, and that's before getting into the crazy things you can do with botany, and  beneficial anomalies (Though dangerous ones are much more common) - Even slimes have their weird esoteric uses from time to time, though I concede that's more like space magic.

So when you say all the interaction "nt really needs" Please consider how diverse the potential interactions can be.  Cooks make food. Robotics makes robots. Medical makes you stay alive. NT, ideally, helps all these do their jobs better, faster, or with more variety. The role of R&D is to research and develop, and ideally the developments can be applied to create interaction and improve the lives / rounds of other characters/players. 

Shunting the department around, and treating players who enjoy it with contempt, are not the makings of an implementable set of ideas that I would want to support under any circumstances.

Neon1ight commented:
My main reason for supporting this change, personally, is the belief that the Expeditionary Corps /should/ have a place in the research department. By moving NanoTrasen to a ship that they own and control independently from command, and is even more free from command oversight than the department, while opening up some science positions (like xenobotany and xenobiology) to Expeditionary Corps personnel (who would work in the Torch labs, not the Petrov) would not only allow NanoTrasen to act as they do now, uncompromised by the ways that made previous attempts unsuccessful, it would also allow EC to be research workers in the labs aboard the Torch itself. Personally, I don't think it makes sense that the EC not be allowed to do science aboard their own ship-- Considering one of it's primary purposes /is/ to do science, especially in the field of xenolife and other such relevant disciplines.  Of course, not the point of this PR, but it's possible.

Faustico commented:
Wait, so is the entire point of this PR to lay the groundwork for an EC research division on the Torch? Eventually giving EC the Torch-bound labs and letting NT do their own thing on their isolated ship?
Because what's gonna happen is that the NT ship will have basically no effect on any rounds if that happens, since if someone needs science for something they'll just talk to the Torch labs instead of trying to get the attention of some scientists on a completely separate ship that may or may not even be docked at the time. Then people will slowly start not playing NT science because hey, it turns out people might want to RP on an HRP server and having to re-dock the ship, walk through an airlock and across a hangar just to say 'hi' to anyone from another department /might/ make that a bit difficult (or at least annoying). Then since no one's playing NT science anymore you might as well get rid of it and give the labs to EC right?

So, yeah, if this gets approved then the slow decline of NT science will start due to a change that adds /nothing/ positive for the people who actually play research.

Neon1ight commented:
@Faustico I don't believe that's the point of this, it's just a possibility that I'd find interesting.

babydoll commented:
Merge conflicts.

This has implications beyond a simple remap and will be carefully considered - **please keep all further comments relating to the PR itself.**

Dopupureganjiru commented:
As far as renovating the petrov, instead of an enormous change, maybe just replace the toxin chamber room with something more useful, like a small xenoflora lab (it already has two trays in one of the ventable chambers, but no other botany equipment), or maybe a partial fabrication lab (analysis only?). The petrov is where you want to put dangerous things and its equipment should probably reflect this.  Being a ship, itself, is in large part a failsafe. So there's less call for it to be fully kitted out, and more call for those on board being able to gather what data they can from whatever danger exists there, and get home safely once it is collected, if nothing goes wrong - or escape if something does. 

Maybe even a science focused medical facility (body scanners, the secret clones even? Stuff like that?) There's plenty of space for thought there without gutting and redesigning the entire branch.

Sbotkin commented:
Can we get controversial tag and a forum thread?

CakeQ commented:
@sabiram Conflicts resolved.

nearlyNonexistent commented:
Yeah I really feel like this thing should be given a forum thread adn controversial tag. It's making a massive change to the structure of the game, map, and round.

Also, hi, person who ported inducers here. If you want more NT gadgets for me to port over and give them, tell me ideas! I'm nearlynon#3292 on Discord. I'm all for giving NT more shit to make them useful and interesting and worth playing.

babydoll commented:
This will be thrown to the forums if necessary after code review is complete

babydoll commented:
Reminder that attempting to compel developers to label something contentious is against the [code of conduct](https://github.com/Baystation12/Baystation12/blob/dev/.github/CODE_OF_CONDUCT.md). There is much more useful feedback to provide than "contenttious????"

Sbotkin commented:
>Reminder that attempting to compel developers to label something contentious is against the code of conduct. There is much more useful feedback to provide than "contenttious????"

You shut down feedback related to this PR (because it's more than just a simple remap) without a dedicated thread, but the dedicated thread won't appear without contentious tag.

CakeQ commented:
This PR is done
only issue is there's an air vent in space somewhere and I honest to god can't find it

Textor44 commented:
I still don't see the point of creating a massive, half-finished room on deck 1 off the main hall. This isn't maintenance or something hidden off the hangar.

comma commented:
Well, other alternative is putting space turfs in there, so I think derelict room is preferable. 
Can expand part with destroyers etc a bit since it's super cramped 1 tile hallway rn.
Maybe add some biggish empty IC testing area, some assemblies could use bit of space for testing, like throwers or mobile ones

comma commented:
Something like this
![image](https://user-images.githubusercontent.com/1300258/45148237-80ad7c80-b1cf-11e8-89fd-8b2585927120.png)

CakeQ committed:
Petrov 2.0 final pass

adds testing chamber

Petrov 2.0 final pass

CakeQ commented:
Updated pics in OP

Textor44 commented:
That's much better. I really just didn't want one of the primary and busiest decks to end up with a large empty room for no good reason.

Textor44 commented:
edit: rephrasing to come off less harshly.

Why are the escape pods on the ~~petrov~~ Torch full of cryo pods? People need to be able to sit down during evac, and I don't think anyone wants to stick themselves in pods to make themselves fall asleep 5 minutes before round end.

CakeQ commented:
@Textor44 makes more sense for a deep space exploration ship innit

Textor44 commented:
@CakeQ It makes them crowded deathtraps. I do not want to go in an escape pod with no chairs.

comma commented:
if you're evacuating in very close proximity of possible rescue , cryopod can be used as very comfy reclining chair
otherwise, chair cannot be used as any kind of cryopod, and you'll die (but sitting)

Textor44 commented:
@comma From a lore perspective, I suppose it makes more sense. From a gameplay perspective, I think this is a bad move.

CakeQ commented:
@Textor44 cryopods are literally luxury chairs, you climb into them before the shuttle launches, easy

CakeQ commented:
wrong button ugh thanks @Textor44

afterthought2 commented:
But you'll then cryo out, removing any RP opportunities available. There's often some amount of trying to stabilize wounded people at the last moment, freaking out about getting away, and generally interacting with a large group of people which you don't get during the regular shift that much that makes escape pods special.

I get it that it makes sense, but it's just not a positive gameplay change. Could make them dummy/cheap versions that don't actually put you under (work slowly, etc) as some kind of compromise.

Textor44 commented:
@afterthought2 while that's an OK compromise, I like the general feeling of seeing who you are talking to. Seeing 7 green pods talking to each other is visually boring, too.

comma commented:
Ok judgement reached by a genus and nice smelling non-biased party
@CakeQ chairs indoot look nicer, pls swap back. Maybe 4 chairs / 3 pods

CakeQ committed:
Gives escape pods their stinky chairs back

Connects Petrov to the Hangar subgrid

Fixes seperated areas

fixes duplicate escape pod controller

fixes ERT landmark

changes ID program requirements

moves torch-specific define to torch files

Last tweaks

fixes piano path

fixes missing pipe

CakeQ commented:
Gave escape pods their chairs back, connected petrov to hangar subgrid

babydoll commented:
Only the captain or HoP should be modifying non-research IDs.

CakeQ commented:
@sabiram Changed to reflect this.

babydoll commented:
@CakeQ 
Take it to the forums. Not sure if there's any possible resolution but it should be discussed anyway.

CakeQ commented:
[Thread](https://forums.baystation12.net/threads/petrov-2-0.6965)

comma commented on code/game/jobs/access_datum.dm:
```diff
@@ -465,13 +465,13 @@
 	access_type = ACCESS_TYPE_NONE
 
-/var/const/access_merchant = 201
+/var/const/access_merchant = 301
```
did you check that this doesn't break mapped in accesses? Like air alarms on merchant ship etc.

CakeQ replied:
Yeah I remapped the merchant base as necessary

comma commented on code/modules/shuttles/shuttle.dm:
```diff
@@ -195,6 +195,8 @@
 			for(var/turf/TD in A.contents)
 				var/turf/TA = GetAbove(TD)
 				if(istype(TA, get_base_turf_by_area(TA)) || istype(TA, /turf/simulated/open))
+					if(get_area(TA) in shuttle_area)
```
What's this?

EDIT: nvm preventing making ceilings inside shuttles over stairs etc

babydoll commented:
The RD's main office should be on the bridge with the rest of execs

CakeQ commented:
@sabiram Completely disagree, the RD should be on the Petrov where he can oversee the labs regardless of if they're docked or undocked, whereas the NTL is the role that would make more sense having an office on the Torch.
I think if you really disagree on this front it might be a good idea to see what other people think on the matter in the forums.

babydoll commented on code/modules/modular_computers/file_system/programs/command/card.dm:
```diff
@@ -132,6 +131,9 @@
 			else
 				module.show_assignments = 1
 		if("print")
+			if(!(access_change_ids in user_id_card.access))
```
This is reused several times and could be a proc

babydoll commented:
The changelog must be much more descriptive
What is the phoron sublimation lab supposed to allow you to do?
Bomb lab has no tables
Grave lack of chairs or anything for this workplace
Areas with solar wires are using wrong floor type, should be `/turf/simulated/floor/airless`
The guard doesn't need 3 lockers

babydoll commented:
Merge conflicts

CakeQ commented:
Addressed all of your concerns @sabiram 
Think the mapbot might be borken but most offices in the petrov have a chair
Phoron lab at the moment is just the a misc lab with a fancy name but I plan to change it with my other PR that's up (phoron desub).

babydoll commented:
If you want the RD to have some kind of office on the shuttle that's fine, it can even be their main office, but they also must have somewhere to work on the bridge.

You should also remove the NT decal from the shuttle foyer since that's no longer very relevant. You can get rid of it from the department foyer, as well.

babydoll commented:
~~Also: the guard still has 3 lockers, there are no tables in toxins still, and the tiles are the wrong type on the solars area.~~
Branch was out of date

babydoll commented:
This can be merged once the RD has that office and the conflicts are resolved.
e: We're keeping this label on for record-keeping, I forgot

babydoll commented:
Author has stated they will not be updating this PR
